### DISCUSSION TITLE
[vite-plugin-commonjs]In curiosity, vite already has optimizeDeps，why do we need this plugin

### DISCUSSION THREAD

**Discussion by Viggo95** [originjs/origin.js#95] [Ideas]
@ygj6 
Sorry, I really don't understand the reason for doing this, so I would like to consult you

**Comment by ygj6** [comment]
optimizeDeps mainly deals with upstream dependencies, and some projects have commonJS syntax in the source code, such as the original webpack project switching to vite, for example, with link dependencies on other local modules, through the vite-plugin-commonjs plugin they can choose to not to modify the source code.

Another part of the reason is that there are still projects configured with optimizeDeps that don't work properly, I don't have statistics on which scenarios, there are examples of such in vite issues.

**Reply by Viggo95** [reply]
Thank you for your answer  and Thank you for this package


I am currently using this package and have encountered the following problem.

- https://github.com/evanw/esbuild/issues/442
- https://github.com/evanw/esbuild/issues/744


At the moment I'm solving it by `@babel/core  transformSync` and then manually adding `export {xxx}` in the file end



If I can verify on my side that it can cover all case, I can provide you with a pr

**Reply by ygj6** [reply]
We very much welcome your PR and wish you good luck.
